packages/terra-data-grid/src/DataGrid.jsx

```diff
@@ -578,9 +608,9 @@ const DataGrid = forwardRef((props, ref) => {
 
         // Check for last focused column ID. If found set the index. Otherwise set it to the last focused column or last index.
         if (focusedCellRef.current.columnId) {
-          newColumnIndex = displayedColumns.findIndex(column => column.id === focusedCellRef.current.columnId);
+          newColumnIndex = tableColumns.findIndex(column => (column.id === focusedCellRef.current.columnId && (column.columnSpanIndex && (column.columnSpanIndex === focusedCellRef.current.columnSpanIndex))));
```

Comment:
This condition is not correct.  You will not always have a column span index.  The condition should account for that properly.  You are just getting by due to the fail safe on the next line.

Reply:
Updated.